source/sundials/ida.cc

```diff
@@ -151,11 +159,79 @@ namespace SUNDIALS
 
       return err;
     }
+#  else
+
+
+
+    template <typename VectorType>
+    int
+    t_dae_jacobian_setup(realtype tt,
+                         realtype cj,
+                         N_Vector yy,
+                         N_Vector yp,
+                         N_Vector /* ignored */,
+                         SUNMatrix /* ignored */,
+                         void *user_data,
+                         N_Vector /* ignored */,
+                         N_Vector /* ignored */,
+                         N_Vector /* ignored */)
+    {
+      Assert(user_data != nullptr, ExcInternalError());
+      IDA<VectorType> &solver = *static_cast<IDA<VectorType> *>(user_data);
+      std::cout << "setting jacobian" << std::endl;
+      GrowingVectorMemory<VectorType> mem;
+
+      typename VectorMemory<VectorType>::Pointer src_yy(mem);
+      solver.reinit_vector(*src_yy);
+
+      typename VectorMemory<VectorType>::Pointer src_yp(mem);
+      solver.reinit_vector(*src_yp);
+
+      copy(*src_yy, yy);
+      copy(*src_yp, yp);
+
+
+      int err = solver.setup_jacobian(tt, *src_yy, *src_yp, cj);
+
+
+      return err;
+    }
+
+
+
+    template <typename VectorType>
+    int
+    t_dae_solve(SUNLinearSolver LS,
+                SUNMatrix /*ignored*/,
+                N_Vector x,
+                N_Vector b,
+                realtype /*tol*/)
```

Comment:
same here with the tolerance. You can probably just copy the same structure from my last KINSOL patch (#12031).

Reply:
I tried to copy your structure. The only difference is that I check if the user has implemented the new version or the deprecated one. If non of the two is implemented a throw an assertion